### PR TITLE
[Reviewer: Graeme] Wildcard IMPU stress testing

### PR DESCRIPTION
**Pull request by rkday-pro**
wildcard_caller.xml is just caller.xml with a P-Profile-Key.

### CODE REVIEW

File: clearwater-sip-stress-coreonly.root/usr/share/clearwater/bin/run_stress [hunk 1]
```diff
@@ -144,14 +145,27 @@ shuffle(callees)
...
+# For wildcard IMPU testing, we expect wildcarded IMPUs to prefix the
+# wildcarded part with a specific character. This makes it easy to mock out
+# HSSes and other components which recognise that these are wildcards.
+WILDCARD_DELIMITER = "$"
+
 register_csv.write("SEQUENTIAL\n")
 for s in registers:
-    register_csv.write("{}\n".format(s))
+    if args.wildcards:
+        register_csv.write("{}{}000\n".format(s, WILDCARD_DELIMITER))
```
Comment (graemerobertson) [inline clearwater-sip-stress-coreonly.root/usr/share/clearwater/bin/run_stress:161]: No?

File: clearwater-sip-stress-coreonly.root/usr/share/clearwater/bin/run_stress [hunk 2]
```diff
@@ -199,6 +213,9 @@ re_reg_cmd = [SIPP_BIN,
 if args.messages:
     caller_file = "message_sender.xml"
     callee_file = "error_message_receiver.xml" if args.with_errors else "message_receiver.xml"
+elif args.wildcards:
+    caller_file = "caller.xml"
```
Comment (graemerobertson) [inline clearwater-sip-stress-coreonly.root/usr/share/clearwater/bin/run_stress]: Is this supposed to be `wildcard_caller.xml`?!

File: clearwater-sip-stress-coreonly.root/usr/share/clearwater/bin/run_stress [hunk 3]
```diff
@@ -199,6 +213,9 @@ re_reg_cmd = [SIPP_BIN,
...
+    callee_file = "wildcard_callee.xml"
```
Comment (graemerobertson) [inline clearwater-sip-stress-coreonly.root/usr/share/clearwater/bin/run_stress]: And where is `wildcard_callee.xml`? This file would just eb the same as the normal file, right?

File: clearwater-sip-stress-coreonly.root/usr/share/clearwater/bin/run_stress (hunk 2 above)
Comment (graemerobertson) [reply clearwater-sip-stress-coreonly.root/usr/share/clearwater/bin/run_stress]: Do we actually need `wildcard_caller.xml`, given it's the same, but with a P-Profile-Key? Can't we just have [field4] as `P-Profile-Key: <blah>\n` or whatever and use the same file?

**Review by graemerobertson — CHANGES_REQUESTED** [review]
Pretty good, apart from the fact that it all looked rubbish to me

File: clearwater-sip-stress-coreonly.root/usr/share/clearwater/bin/run_stress (hunk 1 above)
Comment (graemerobertson) [reply clearwater-sip-stress-coreonly.root/usr/share/clearwater/bin/run_stress:161]: Ah, I think I understand this now (having seen the Husk changes) - worth a comment saying that for wildcard badgers Husky expects <blah>$000 as the/a distinct IMPU, hence we use it for registrations?

Comment (rkday-pro) [reply clearwater-sip-stress-coreonly.root/usr/share/clearwater/bin/run_stress:161]: Yep, comment updated.

File: clearwater-sip-stress-coreonly.root/usr/share/clearwater/bin/run_stress (hunk 2 above)
Comment (rkday-pro) [reply clearwater-sip-stress-coreonly.root/usr/share/clearwater/bin/run_stress]: Good idea - done.

File: clearwater-sip-stress-coreonly.root/usr/share/clearwater/bin/run_stress (hunk 3 above)
Comment (rkday-pro) [reply clearwater-sip-stress-coreonly.root/usr/share/clearwater/bin/run_stress]: Yes - I mixed up caller and callee here.

**Pull request merged**